internal/cmd/github-issue-reopener/reopener/reopener.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO(https://github.com/ianlewis/todos/issues/48): open issue
+// TODO(https://github.com/ianlewis/todos/issues/58): open issue
+
+package reopener
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/google/go-github/v52/github"
+
+	"github.com/ianlewis/todos/internal/cmd/github-issue-reopener/options"
+	"github.com/ianlewis/todos/internal/walker"
+)
+
+// labelMatch is the regexp that matches the TODO label. It can be of the
+// following forms:
+// - http://github.com/owner/repo/issues/1234
+// - github.com/owner/repo/issues/1234
+// - #1234
+// - 1234
+var labelMatch = regexp.MustCompile("((https?://)?github.com/(.*)/(.*)/issues/|#?)([0-9]+)")
+
+// IssueRef is a set of references to an issue.
+type IssueRef struct {
+	// ID is the issue ID.
+	ID int
+
+	// TODOs is a list of todos referencing this issue.
+	TODOs []*walker.TODORef
+}
+
+// IssueReopener is reopens issues referenced by TODOs.
+type IssueReopener struct {
+	client *github.Client
+
+	options *options.Options
+
+	// issues is a local cache of issues.
+	issues struct {
+		sync.Mutex
+		cache map[int]*github.Issue
```

Comment:
Only whether the issue is open or not is needed. Could just have a `map[int]bool` here.

Reply:
Actually the issue's state is used so we do need the full issue.